### PR TITLE
[DROOLS-3280] DMNRuntime API for evaluateByID/Name to error if empty …

### PR DESCRIPTION
**Pull request by tkobayas**
…array

Please review. I have two points:

1) I implemented to throw an IllegalArgumentException but if you prefer another Exception or just creating an error message with MsgUtil.reportMessage(), please let me know.

2) I added 2 unit test methods. Is it verbose? But may be better than no test...

### CODE REVIEW

**Review by hellowdan — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/impl/DMNRuntimeImpl.java [hunk 1]
```diff
@@ -188,6 +188,9 @@ public DMNResult evaluateByName( DMNModel model, DMNContext context, String... d
         Objects.requireNonNull(model, () -> MsgUtil.createMessage(Msg.PARAM_CANNOT_BE_NULL, "model"));
         Objects.requireNonNull(context, () -> MsgUtil.createMessage(Msg.PARAM_CANNOT_BE_NULL, "context"));
         Objects.requireNonNull(decisionNames, () -> MsgUtil.createMessage(Msg.PARAM_CANNOT_BE_NULL, "decisionNames"));
+        if (decisionNames.length == 0) {
+            throw new IllegalArgumentException(MsgUtil.createMessage(Msg.PARAM_CANNOT_BE_EMPTY, "decisionNames"));
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/impl/DMNRuntimeImpl.java:192]: 👍 

> I implemented to throw an IllegalArgumentException

it's good, as it is consistent with the RuntimeException thrown by the `Objects.requireNonNull` if the parameters are null.

**Review by tarilabs — APPROVED** [review]
thank you!

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 2]
```diff
@@ -2720,5 +2721,43 @@ public void testBindingContextTypeCheck() {
...
+
+        final DMNContext context = DMNFactory.newContext();
+        context.set( "Monthly Salary", 1000 );
+
+        try {
+            String[] decisionNames = new String[]{};
+            runtime.evaluateByName(dmnModel, context, decisionNames);
+            fail("Should throw an IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+            assertTrue(true);
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: You can annotate the test method with @Test(expected = IllegalArgumentException.class). That way you can remove the try-catch block and the fail(...) call.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 3]
```diff
@@ -2720,5 +2721,43 @@ public void testBindingContextTypeCheck() {
...
+
+        final DMNContext context = DMNFactory.newContext();
+        context.set( "Monthly Salary", 1000 );
+
+        try {
+            String[] decisionIds = new String[]{};
+            runtime.evaluateById(dmnModel, context, decisionIds);
+            fail("Should throw an IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+            assertTrue(true);
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Same as my previous comment.

**Review by baldimir — CHANGES_REQUESTED** [review]

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java (hunk 2 above)
Comment (tkobayas) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Fixed. Thanks!

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java (hunk 3 above)
Comment (tkobayas) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Fixed. Thanks!

**Review by baldimir — APPROVED** [review]

**Review by tarilabs — APPROVED** [review]

**Pull request merged**